### ISSUE TITLE
Application not working

### ISSUE THREAD

**Issue by Amitosh1** [krakenjs/kraken-example-with-specialization#2]
while trying url like

http://localhost:8000/setSpcl/yin
(or)
http://localhost:8000/setSpcl/yang

I am getting exception
![capture](https://cloud.githubusercontent.com/assets/6291256/8258362/f1ac3dda-1679-11e5-9a42-52e4c91a7b77.PNG)

**Comment by aredridel** [comment]
Are you running in prod or dev mode?

If prod, have you run the build with `grunt build`?

This usually means 'template not found', which often means it wasn't built.

**Comment by Amitosh1** [comment]
I am running this on dev mode.

I tried changing to prod mode also.
But there also while hitting yin or yang url main page is loading as
yinyang but more info link worked properly

**Comment by grawk** [comment]
I got a much more insightful error in production mode:

``` shell
TypeError: Cannot read property 'substr' of undefined
    at peg$parseraw (/Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/dustjs-linkedin/lib/parser.js:2100:16)
    at peg$parsepart (/Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/dustjs-linkedin/lib/parser.js:401:12)
    at peg$parsebody (/Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/dustjs-linkedin/lib/parser.js:384:12)
    at peg$parsestart (/Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/dustjs-linkedin/lib/parser.js:374:12)
    at parse (/Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/dustjs-linkedin/lib/parser.js:2639:18)
    at Object.compiler.compile (/Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/dustjs-linkedin/lib/compiler.js:26:27)
    at /Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/adaro/lib/patch/index.js:82:64
    at /Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/adaro/lib/reader/js.js:39:13
    at /Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/engine-munger/lib/munger.js:89:13
    at /Users/medelman/src/krakex/kraken-example-with-specialization/node_modules/engine-munger/lib/cache.js:65:13
```

**Comment by grawk** [comment]
Trying to unwind this, but it appears that instead of "yin" there is a corrupted compiled template called "yinyangyang" that gets added to cache.
![yinyangyang](https://cloud.githubusercontent.com/assets/1031360/8258722/105a1e7c-166c-11e5-9085-cd24607e7ef3.png)

**Comment by aredridel** [comment]
It may be clearer with dust 2.5, which was the last one that worked perfectly with specialization.

**Comment by aredridel** [comment]
Looks like the changes to https://github.com/krakenjs/engine-munger/blob/v0.2.x/lib/munger.js#L83 are to blame. The template name is mangled twice, because that code tries to be compatible with both old and new dust.

**Comment by aredridel** [comment]
This is what we get for editing an AST as a string.

**Comment by aredridel** [comment]
Fixed by https://github.com/krakenjs/engine-munger/issues/34 and published as `engine-munger@0.2.8`
